.gitignore

```diff
@@ -1,33 +1,68 @@
-# Compiled class file
-*.class
-
-# Log file
-*.log
-
-# BlueJ files
-*.ctxt
-
-# Mobile Tools for Java (J2ME)
+# Java
 .mtj.tmp/
-
-# Package Files #
+*.class
 *.jar
 *.war
-*.nar
 *.ear
-*.zip
-*.tar.gz
-*.rar
-
-# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
+*.nar
 hs_err_pid*
 
+# Maven
+target/
+pom.xml.tag
+pom.xml.releaseBackup
+pom.xml.versionsBackup
+pom.xml.next
+pom.xml.bak
+release.properties
+dependency-reduced-pom.xml
+buildNumber.properties
+.mvn/timing.properties
+.mvn/wrapper/maven-wrapper.jar
 
-.vscode
-.DS_store
-.gradle/*
+# Gradle
+bin/
+build/
 .gradle
-build
-.settings
+.gradletasknamecache
+gradle-app.setting
+!gradle-wrapper.jar
+
+# IntelliJ
+out/
+.idea/
+.idea_modules/
+*.iml
+*.ipr
+*.iws
+
+# Eclipse
+.settings/
+bin/
+tmp/
+.metadata
+.classpath
+.project
+*.tmp
+*.bak
+*.swp
+*~.nib
+local.properties
+.loadpath
+.factorypath
+
+# NetBeans
+nbproject/private/
+build/
+nbbuild/
+dist/
+nbdist/
+nbactions.xml
+nb-configuration.xml
+
+# Visual Studio Code
+.vscode/
+.code-workspace
 
-gradle.properties
```

Comment:
Let's leave gradle.properties in the gitignore. It does expose our ossrh account and signing password as well

Reply:
We should move these credentials into 'repo secrets' when we set up CI/CD. This file usually contains properties for builds, so I think we need it. Let's just avoid committing credentials here for now. Do you agree?